library/src/main/java/com/wultra/android/mtokensdk/push/PushParser.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020, Wultra s.r.o. (www.wultra.com).
+ *
+ * All rights reserved. This source code can be used only for purposes specified
+ * by the given license contract signed by the rightful deputy of Wultra s.r.o.
+ * This source code can be used only by the owner of the license.
+ *
+ * Any disputes arising in respect of this agreement (license) shall be brought
+ * before the Municipal Court of Prague.
+ */
+
+package com.wultra.android.mtokensdk.push
+
+/**
+ * Helper class that can translate incoming notifications from FCM to [PushMessage].
+ */
+class PushParser {
+
+    companion object {
+
+        /**
+         * When you receive a push notification, you can test it here if it's a "WMT" notification.
+         *
+         * Return type can be [PushMessageOperationCreated] or [PushMessageOperationFinished]
+         *
+         * @param notificationData: data of received notification.
+         * @return parsed known [PushMessage] or null
+         */
+        @JvmStatic
+        @JvmName("parseNotification")
+        fun parseNotification(notificationData: Map<String, String>): PushMessage? {
+
+            val id = notificationData["operationId"] ?: return null
+            val name = notificationData["operationName"] ?: return null
+
+            return when (notificationData["messageType"]) {
```

Comment:
The method should also return any other data that push message sender decides to include in the payload, for example as an `extras` attributes.

Reply:
Yea, I added an `originalData` property to it.